ui-tests/src/test/java/io/syndesis/qe/steps/other/ODataSteps.java

```diff
@@ -0,0 +1,76 @@
+package io.syndesis.qe.steps.other;
+
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import io.cucumber.datatable.DataTable;
+import io.syndesis.qe.accounts.Account;
+import io.syndesis.qe.accounts.AccountsDirectory;
+import io.syndesis.qe.utils.HTTPResponse;
+import io.syndesis.qe.utils.HttpUtils;
+import io.syndesis.qe.utils.ODataUtils;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Headers;
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+public class ODataSteps {
+
+    @When("^create OData credentials$")
+    public void createODataCredentials() {
+        Account oData = new Account();
+        oData.setService("OData");
+        Map<String, String> properties = new HashMap<>();
+        properties.put("serviceUri", ODataUtils.getOpenshiftService());
+        oData.setProperties(properties);
+        AccountsDirectory.getInstance().addAccount("odata", oData);
+        log.info("Created new Account: odata");
+    }
+
+    @When("^.*insert entity \"([^\"]*)\" into \"([^\"]*)\" collection on OData service$")
+    public void insertEntityToODataService(final String entity, final String collection) {
+        String requestBody = ODataUtils.readResourceFile(this.getClass().getClassLoader().getResource(entity));
+        Headers headers = Headers.of("Content-Type", "application/json");
+        HTTPResponse response = HttpUtils.doPostRequest(ODataUtils.getOpenshiftRoute() + collection, requestBody, headers);
+        assertThat(response.getCode()).isEqualTo(201);
+        log.info("Entity from resource file " + entity + " succesfully inserted to sample OData service");
+    }
+
+    @Then("^.*check that \"([^\"]*)\" entity in \"([^\"]*)\" collection contains$")
+    public void oDataServiceContains(final String entityKey, final String collection, DataTable table) {
+        // Currently supports only entityKey in number format
+        HTTPResponse response = HttpUtils.doGetRequest(ODataUtils.getOpenshiftRoute() + collection + "(" + entityKey + ")");
+        assertThat(response.getCode()).isEqualTo(200);
+        String json = new JSONObject(response.getBody()).toString();
```

Comment:
Aaaah, thats a line I left there by mistake (previously I converted it to map and was checking the key:value pairs in that map, then figured out I can do the same with JSONObject, but there was a problem with types - datatable takes only strings, so I thought it will be easier to just check if the response body contains the values needed).
Even though the method doesn't expect two columns with key:value pairs, it can be given it this way and the method should check if it is in that json (only it doesn't check if the key is really a key and the value is really a value).